File: app/src/main/java/com/vo1d/schedulemanager/v2/ui/schedule/DaysAdapter.java
package com.vo1d.schedulemanager.v2.ui.schedule;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.vo1d.schedulemanager.v2.data.days.Day;
import com.vo1d.schedulemanager.v2.ui.classes.list.ClassesListFragment;

import java.util.LinkedList;
import java.util.List;

class DaysAdapter extends FragmentStateAdapter {
    private final List<Integer> ids = new LinkedList<>();
    private List<Day> days = new LinkedList<>();

    public DaysAdapter(@NonNull Fragment fragment) {
        super(fragment);
    }

    @Override
    public boolean containsItem(long itemId) {
        return ids.contains((int) itemId);
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        int id = days.get(position).id;
        return new ClassesListFragment(id);
    }

    @Override
    public long getItemId(int position) {
        return ids.get(position);
    }

    @Override
    public int getItemCount() {
        return days.size();
    }

    public void submitList(List<Day> days) {
        if (days != this.days) {
            this.days = days;
            ids.clear();
            for (Day day : this.days) {
                ids.add(day.id);
            }
            notifyDataSetChanged();
        }
    }
}

File: app/src/main/kotlin/ru/vo1d/timetablemanager/ui/sections/timetable/DaysAdapter.kt
package ru.vo1d.timetablemanager.ui.sections.timetable

import android.annotation.SuppressLint
import androidx.fragment.app.Fragment
import androidx.viewpager2.adapter.FragmentStateAdapter
import kotlinx.datetime.DayOfWeek
import kotlinx.datetime.isoDayNumber
import ru.vo1d.timetablemanager.data.DatabaseEntity.Companion.INVALID_ID
import ru.vo1d.timetablemanager.ui.sections.timetable.sessions.list.SessionsListFragment

@SuppressLint("NotifyDataSetChanged")
internal class DaysAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
    private var days = emptyList<DayOfWeek>()
    private var weekId = INVALID_ID

    override fun containsItem(itemId: Long) =
        days.contains(DayOfWeek.of(itemId.toInt()))

    override fun getItemId(position: Int) =
        days[position].isoDayNumber.toLong()

    override fun createFragment(position: Int) =
        SessionsListFragment(weekId, days[position])

    override fun getItemCount() =
        days.size


    fun submitList(list: List<DayOfWeek>) {
        this.days = list
        notifyDataSetChanged()
    }

    fun setWeekId(weekId: Int) {
        this.weekId = weekId
        notifyDataSetChanged()
    }

    fun getItemAt(position: Int) =
        days.getOrNull(position)
}